### PR TITLE
Adding e2e test cases for auto-approval domains

### PR DESCRIPTION
**Pull request by dperaza4dustbit**
Also adding new host wait conditions for config changes and some extra make targets to improve the time to reset operators on dev environment

paired with https://github.com/codeready-toolchain/host-operator/pull/1019

### CODE REVIEW

File: make/test.mk [hunk 1]
```diff
@@ -50,6 +50,16 @@ test-e2e: prepare-e2e verify-migration-and-deploy-e2e e2e-run-parallel e2e-run
 test-e2e-without-migration: prepare-e2e deploy-e2e e2e-run-parallel e2e-run
 	@echo "To clean the cluster run 'make clean-e2e-resources'"
 
+.PHONY: test-e2e-no-parallel
+## Run the e2e tests without migration tests
```
Comment (alexeykazakov) [inline make/test.mk:54]: Copy-paste from a different target. It seems that this is actually skipping both parallel and migration tests.

File: make/test.mk [hunk 2]
```diff
@@ -93,6 +103,16 @@ label-olm-ns:
 test-e2e-local-without-migration:
 	$(MAKE) test-e2e-without-migration HOST_REPO_PATH=${PWD}/../host-operator MEMBER_REPO_PATH=${PWD}/../member-operator REG_REPO_PATH=${PWD}/../registration-service
 
+.PHONY: test-e2e-local-no-parallel
+## Run the e2e tests with the local 'host', 'member', and 'registration-service' repositories but without migration tests
```
Comment (alexeykazakov) [inline make/test.mk:107]: Same here?

File: make/test.mk [hunk 3]
```diff
@@ -93,6 +103,16 @@ label-olm-ns:
...
+test-e2e-local-no-parallel:
+	$(MAKE) test-e2e-no-parallel HOST_REPO_PATH=${PWD}/../host-operator MEMBER_REPO_PATH=${PWD}/../member-operator REG_REPO_PATH=${PWD}/../registration-service
+
+.PHONY: prepare-and-deploy-e2e-local
+## Run the e2e tests with the local 'host', 'member', and 'registration-service' repositories but without migration tests
```
Comment (alexeykazakov) [inline make/test.mk:112]: And here.

**Review by alexeykazakov — COMMENTED** [review]
Looks good overall. Just a few minor comments.

**Comment by alexeykazakov** [comment]
Also, please do not forget to add links to the paired PR in the description. I did it for you though.

File: make/test.mk (hunk 1 above)
Comment (dperaza4dustbit) [reply make/test.mk]: fixed

File: make/test.mk (hunk 2 above)
Comment (dperaza4dustbit) [reply make/test.mk]: Fixed

File: make/test.mk (hunk 3 above)
Comment (dperaza4dustbit) [reply make/test.mk]: Fixed

**Comment by dperaza4dustbit** [comment]
> Also, please do not forget to add links to the paired PR in the description. It helps with reviewing PRs. I did it for you though.

Got it

**Review by alexeykazakov — APPROVED** [review]

File: make/test.mk [hunk 4]
```diff
@@ -50,6 +50,16 @@ test-e2e: prepare-e2e verify-migration-and-deploy-e2e e2e-run-parallel e2e-run
 test-e2e-without-migration: prepare-e2e deploy-e2e e2e-run-parallel e2e-run
 	@echo "To clean the cluster run 'make clean-e2e-resources'"
 
+.PHONY: test-e2e-no-parallel
+## Run the e2e tests without migration and without parallel tests
+test-e2e-no-parallel: prepare-e2e deploy-e2e e2e-run
+	@echo "To clean the cluster run 'make clean-e2e-resources'"
```
Comment (MatousJobanek) [inline make/test.mk:56]: wouldn't it be better to name it
`test-e2e-sequential-only`

**Pull request merged**